content_script.js

```diff
@@ -137,38 +140,104 @@ class HTMLSerializer {
    *
    * @param {Element} element The element being processed, which has the src
    *     attribute.
+   * @param {Set<string>} processedAttributes The Set containing all attributes
+   *     already added to |this.html| for the Element.
    * @private
    */
-  processSrcAttribute(element) {
-    if (element.tagName.toLowerCase() != 'iframe') {
-      this.processSrcHole(element);
-    } 
+  processSrcAttribute(element, processedAttributes) {
+    var tag = element.tagName.toLowerCase();
+    switch(tag) {
+      case 'iframe':
+        break; // Do nothing.
+      case 'source':
+        if (!element.parent || element.parent.tagName.toLowerCase() != 'img') {
+          this.processSimpleSrc(element, processedAttributes);
+          break;
+        } // else process as img.
+      case 'input':
+        var type = element.attributes.type;
+        if (tag == 'input' && (!type || type.value.toLowerCase() != 'image')) {
+          break;
+        } // else process as img.
+      case 'img':
+        if (window.location.host == this.fullyQualifiedURL(element).host) {
+          this.processSrcHole(element, processedAttributes);
+          break;
+        }
+      default:
+        this.processSimpleSrc(element, processedAttributes);
+    }
+  }
+
+  /**
+   * Get a URL object for the value of the elements src attribute.
+   *
+   * @param {Element} element The element for which to retrieve the URL.
+   * @return {URL} The URL object.
+   */
+  fullyQualifiedURL(element) {
+    var url = element.attributes.src.value;
+    var a = document.createElement('a');
+    a.href = url;
+    url = a.href; // Retrieve fully qualified URL.
+    return new URL(url);
```

Comment:
From playing around a little, it seems that creating an inline style that specifies a height or width value overwrites the height or width html attribute.  Therefore, I think it may make sense to simply ignore the height and width attributes (process them like any other attribute) because they would be overwritten anyway by the inline style (which I retrieve via document.getComputedStyle).  Even if a src url is not specified, the inline style width and height will cause the <img> element to take up the same amount of space as it would if the src url is specified, which handles the case we were considering.

Let me know what you think.